pkg/registry/core/node/strategy.go

```diff
@@ -82,6 +83,27 @@ func (nodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Objec
 	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) && !nodeConfigSourceInUse(oldNode) {
 		newNode.Spec.ConfigSource = nil
 	}
+	dropDisabledFields(newNode, oldNode)
+}
+
+func dropDisabledFields(node *api.Node, oldNode *api.Node) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && !multiNodeCIDRsInUse(oldNode) {
+		if len(node.Spec.PodCIDRs) > 1 {
```

Comment:
why not call `multiNodeCIDRsInUse` here too?

Reply:
it is style matching across the entire code file (and PodIPs as well), nothing more.